### PR TITLE
Feat: From yaml injection

### PR DESCRIPTION
**Pull request by yue9944882**
simplies resource loading in spring java.

```java
public class Application {
    // #1 injecting fields
    @FromYaml("my-service.yaml") // will look for the yaml file 1. on the machine 2. in the classpath
    private V1Service service;

    @Configuration
    static class MyConfiguration {
       // #2 injecting beans
       @FromYaml("my-service.yaml")
       public V1Service service() {
           return new V1Service();
       }
    }
}
```

### CODE REVIEW

**Comment by yue9944882** [comment]
@ryanjbaxter i suppose this will be also helpful for simplifying the integration tests on the other project.

**Comment by ryanjbaxter** [comment]
@yue9944882 for sure

File: spring/src/main/java/io/kubernetes/client/spring/extended/manifests/KubernetesFromYamlProcessor.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        throw new BeanCreationException(
+            "No such file " + targetFilePath + " either on the machine or classpaths");
+      }
+    }
+
+    try {
+      String yamlContent = new String(Files.readAllBytes(targetPath));
+      Object loadedObj = Yaml.load(yamlContent);
+      return loadedObj;
+    } catch (IOException e) {
```
Comment (brendandburns) [inline spring/src/main/java/io/kubernetes/client/spring/extended/manifests/KubernetesFromYamlProcessor.java:106]: Should we catch a parse error here too?

**Comment by brendandburns** [comment]
Cool stuff, one question about error handling.

File: spring/src/main/java/io/kubernetes/client/spring/extended/manifests/KubernetesFromYamlProcessor.java (hunk 1 above)
Comment (yue9944882) [reply spring/src/main/java/io/kubernetes/client/spring/extended/manifests/KubernetesFromYamlProcessor.java:107]: added catching `YAMLException`, is that right?

**Pull request merged**